src/roles/nuh-deploy/tasks/main.yml

```diff
@@ -132,14 +132,26 @@
           mode: 0640
           owner: "root"
           group: "root"
+        when: custom_configuration_file_location is defined
+
+      - block:
+
+        - name: Add stats-out proxy entries to NUH configuration
+          replace:
+            path: /opt/proxy/data/config.yml
+            regexp: "role: vsdconfig.*$"
+            replace: "role: vsdconfig, enabled: true, firewallports: ['{{ stats_out_proxy_ui_port }}', '{{ stats_out_proxy_api_port }}', '{{ stats_out_proxy_jms_port }}', '{{ stats_out_proxy_xmpp_port }}'], settings: {uiport: '{{ stats_out_proxy_ui_port }}', apiport: '{{ stats_out_proxy_api_port }}', xmppport: '{{ stats_out_proxy_xmpp_port }}', jmsport: '{{ stats_out_proxy_jms_port }}', geo: false}}"
+          when: stats_out_proxy | default('NONE') in [hostname, mgmt_ip]
+
+        when: NUH_DOES_NOT_YET_SUPPORT_STATS_OUT_PROXY | default(false)
 
       - name: Run the configuration script on NUH
         command: ansible-playbook configure.yml
         args:
           chdir: /opt/proxy/bin/ansible
 
     when:
-      - custom_configuration_file_location is defined
+      - custom_configuration_file_location is defined or stats_out_proxy | default('NONE') in [hostname, mgmt_ip]
```

Comment:
Does this needs a - not skip_vsd_installed_check similar to line 124. Should this be run if VSD is not present?

Reply:
I don't know, please advise what this check does.  Note, that the check was not there before for the "custom_configuration_file".  My purpose for this block is to add the stats_out configs to the nuh configuration with or without a custom config present.

Reply:
The skip_vsd_installed_check installs NUH when no VSD is installed. So any tasks that require communication with the VSD are not run when skip_vsd_installed_check is true

Reply:
This is applying the proxy config to the NUH.  It doesn't require communication to the VSDs.  Thus, the check is not needed?

Reply:
Okay, then we do not need a check